### PR TITLE
DOMA-2291 fix contact creation at ticket create page

### PR DESCRIPTION
**Pull request by sitozzz**
change useContactsEditorHook canCreateContact field value from ref to state

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/contact/components/ContactsEditor/index.tsx [hunk 1]
```diff
@@ -149,10 +149,42 @@ export const ContactsEditor: React.FC<IContactEditorProps> = (props) => {
...
         onChange && onChange(contact, isNew)
     }
 
+    const initialValueIsPresentedInFetchedContacts = fetchedContacts && initialValue && initialValue.name && initialValue.phone && find(fetchedContacts, initialValue)
+
+    const sameAsInitial = (contact) => (
+        initialValue && initialValue.name === contact.name && initialValue.phone === contact.phone
+    )
+
+    const isContactSelected = useCallback((contact, i) => {
```
Comment (AntonAL) [inline apps/condo/domains/contact/components/ContactsEditor/index.tsx:163]: Please, rename this constant.
It's name is like a getter-function, but it has side-effect logic inside.

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ export const ContactsInfo = ({ ContactsEditorComponent, form, selectedPropertyId
     const TicketFromResidentMessage = intl.formatMessage({ id: 'pages.condo.ticket.title.TicketFromResident' })
     const TicketNotFromResidentMessage = intl.formatMessage({ id: 'pages.condo.ticket.title.TicketNotFromResident' })
 
-    const contactId = useMemo(() => get(initialValues, ['contact', 'id']), [initialValues])
```
Comment (AntonAL) [inline apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:52]: Are you sure about it?
`contactId` seem to be not an id after you remove getting `id` property.

File: apps/condo/domains/contact/components/ContactsEditor/index.tsx (hunk 1 above)
Comment (sitozzz) [reply apps/condo/domains/contact/components/ContactsEditor/index.tsx]: @AntonAL split `<Option />` selection logic into `useEffect` & cleanup callback

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx (hunk 2 above)
Comment (sitozzz) [reply apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:52]: At `initialValues` object `contact` field is already `id` of this contact

Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:52]: initialValues ​​receives not the contact object, but its id

изображение

**Pull request merged**